services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java

```diff
@@ -84,13 +87,36 @@ public boolean isValid(SubmissionPayload submissionPayload, ConstraintValidatorC
       List<TemporaryExposureKey> exposureKeys = submissionPayload.getKeysList();
       validatorContext.disableDefaultConstraintViolation();
 
-      return checkStartIntervalNumberIsAtMidNight(exposureKeys, validatorContext)
+      boolean isValidPayload = checkStartIntervalNumberIsAtMidNight(exposureKeys, validatorContext)
           && checkKeyCollectionSize(exposureKeys, validatorContext)
           && checkOriginCountryIsValid(submissionPayload, validatorContext)
           && checkVisitedCountriesAreValid(submissionPayload, validatorContext)
           && checkRequiredFieldsNotMissing(exposureKeys, validatorContext)
           && checkTransmissionRiskLevelIsAcceptable(exposureKeys, validatorContext)
           && checkDaysSinceOnsetOfSymptomsIsInRange(exposureKeys, validatorContext);
+
+      if (!isValidPayload) {
+        String payloadString = getPayloadStringFromKeys(submissionPayload.getKeysList());
```

Comment:
What about logging the submission payload including visitedCountries, originCountry and consentToFederation? RequestPadding might be hard, maybe it makes sense to log only the padding length

Reply:
PR changed accordingly.